src/components/Settings/Settings.js

```diff
@@ -30,18 +42,122 @@ export default class Settings extends React.Component {
             <option value="TestNet">TestNet</option>
             <option value="CozNet">CozNet</option>
             <option value="nOSLocal">nOS Local</option>
+
+            {
+              this.props.allNetworks.map((network) => {
+                return (
+                  <option
+                    key={network.neoscan}
+                    value={network.neoscan}
+                  >
+                    {network.name}
+                  </option>
+                );
+              })
+            }
           </select>
         </label>
+        <div className={styles.buttonContainer}>
+          Custom Neoscan Url: {this.props.currentNetwork}
+        </div>
+        {this.renderButtons()}
         {this.state.saved && <Saved /> }
       </div>
     );
   }
 
+  renderButtons = () => {
+    return (
+      <div>
+        <Button onClick={this.handleAddNewNetwork}>
+          {'Add custom network configuration'}
+        </Button>
+        <div className={styles.divider} />
+        <Button onClick={this.handleClearNetwork}>
+          {'Clear custom network configurations'}
+        </Button>
+      </div>
+    );
+  }
+
+  handleClearNetwork = () => {
+    this.props.clearNetworks();
+    this.props.setCurrentNetwork('TestNet');
+    this.props.alert('All custom network configurations cleared');
+  }
+
+  handleAddNewNetwork = () => {
+    this.props.confirm((
+      <div>
+        <Input
+          id="networkName"
+          type="text"
+          label="Network name"
+          placeholder="Network name"
+          onChange={this.handleChangeNetworkName}
+        />
+        <Input
+          id="networkURL"
+          type="text"
+          label="Network URL"
+          placeholder="Network URL"
+          onChange={this.handleChangeNetworkUrl}
+        />
+      </div>
+    ), {
+      title: 'New network configuration',
+      onConfirm: this.handleConfirmAddNetwork,
+      onCancel: noop
+    });
+  }
+
+  handleConfirmAddNetwork = () => {
+    const network = this.props.allNetworks.find((element) => {
+      return element.neoscan === this.props.networkUrl;
+    });
+
+    if (network) {
+      this.props.alert('Error: A network configuration with that Url already exist');
+      return;
+    }
+
+    const newNetwork = { name: this.props.networkName, neoscan: this.props.networkUrl };
+    this.props.addNetwork(newNetwork);
+    this.props.setCurrentNetwork(this.props.networkUrl);
```

Comment:
We've been storing the network as "MainNet", "TestNet", etc. previously.  It looks like this will pass in a URL instead now.  I think we should pass in the name instead.  I think this aligns with all my other name-related comments.

Reply:
Previously I have experimented with this
```
 export default function withNetworkData(propName = 'net') {
   const mapSettingsDataToProps = (currentNetwork) => ({
-    name: currentNetwork.name, [propName]: currentNetwork.neoscan
+    [propName]: currentNetwork.neoscan
   });
   return withData(currentNetworkActions, mapSettingsDataToProps);
 }

```

Reply:
We can either allow the user to provide neoscan URL as part of their config, or we can just point neoscan to "http://localhost:4000/api/main_net" maybe as a starting point.  I don't think we have to dynamically pass in the network name to neoscan since that won't work.

Reply:
For the first option, does that mean that the user cannot add new neoscan URL after their config? (at the beginning) and for the second option, it feels like its actually not completing the PR, and in fact im not sure how we can point neoscan to "http://localhost:4000/api/main_net"  without saving the url of "http://localhost:4000/api/main_net" .

Reply:
@mhuggins if I have to save the network name instead of the URL, then i might have to do the mapping from name to URL somewhere in currentNetworkActions.js, would that be okay? I'll code up an implementation after work.